### PR TITLE
[Backport] exclude remote models in circuit breaker checks and fix memory CB bugs

### PR DESCRIPTION
**Pull request by Zhangxunmt**
### Description
Manual backport of https://github.com/opensearch-project/ml-commons/pull/2469
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ylwu-amzn** [comment]
Make sure CI can pass

File: plugin/src/test/java/org/opensearch/ml/model/MLModelManagerTests.java [hunk 1]
```diff
@@ -451,21 +451,32 @@ public void testRegisterMLRemoteModel() throws PrivilegedActionException {
...
+        when(indexResponse.getId()).thenReturn("mockIndexId");
         modelManager.registerMLRemoteModel(pretrainedInput, pretrainedTask, listener);
 
-        ArgumentCaptor<Exception> argCaptor = ArgumentCaptor.forClass(Exception.class);
-        verify(listener, times(1)).onFailure(argCaptor.capture());
-        Exception e = argCaptor.getValue();
-        assertTrue(e instanceof MLLimitExceededException);
-        assertEquals(memCBIsOpenMessage, e.getMessage());
+        assertEquals(pretrainedTask.getFunctionName(), FunctionName.REMOTE);
+        verify(mlTaskManager).updateMLTask(anyString(), anyMap(), anyLong(), anyBoolean());
```
Comment (dhrubo-os) [inline plugin/src/test/java/org/opensearch/ml/model/MLModelManagerTests.java:479]: This verify is verifying the method is being invoked. But I think what we want to verify here is the success response. I doubt this test is actually serving the purpose we want.

File: plugin/src/main/java/org/opensearch/ml/breaker/MemoryCircuitBreaker.java [hunk 2]
```diff
@@ -50,6 +50,6 @@ public Short getThreshold() {
 
     @Override
     public boolean isOpen() {
-        return jvmService.stats().getMem().getHeapUsedPercent() > this.getThreshold();
+        return getThreshold() < 100 && jvmService.stats().getMem().getHeapUsedPercent() > getThreshold();
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/breaker/MemoryCircuitBreaker.java:53]: For better readability, we should put this `100` in a static variable and add some comment accordingly.

**Comment by Zhangxunmt** [comment]
> Make sure CI can pass

/__e/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)

It's failing due to the github machine running CI lack glibc dependencies. Verified the CI manually from local machine though.

**Review by dhrubo-os — DISMISSED** [review]

**Comment by ylwu-amzn** [comment]
> > Make sure CI can pass
> 
> /__e/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /__e/node20/bin/node) /__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)
> 
> It's failing due to the github machine running CI lack glibc dependencies. Verified the CI manually from local machine though.

This PR https://github.com/opensearch-project/ml-commons/pull/2627 is a temp solution to fix the error. 
@peterzhuamazon, how about backport that fix to old release branch ? So we can verify the patch with CI.

**Comment by Zhangxunmt** [comment]
> > > Make sure CI can pass
> > 
> > 
> > /__e/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /__e/node20/bin/node) /__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)
> > It's failing due to the github machine running CI lack glibc dependencies. Verified the CI manually from local machine though.
> 
> This PR #2627 is a temp solution to fix the error. @peterzhuamazon, how about backport that fix to old release branch ? So we can verify the patch with CI.

I manually added PR 2627 in this commit but still not working for Java in Linux. Can you pls check @peterzhuamazon

**Comment by peterzhuamazon** [comment]
> > > > Make sure CI can pass
> > > 
> > > 
> > > /__e/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /__e/node20/bin/node) /__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)
> > > It's failing due to the github machine running CI lack glibc dependencies. Verified the CI manually from local machine though.
> > 
> > 
> > This PR #2627 is a temp solution to fix the error. @peterzhuamazon, how about backport that fix to old release branch ? So we can verify the patch with CI.
> 
> I manually added PR 2627 in this commit but still not working for Java in Linux. Can you pls check @peterzhuamazon

Like I said in the original PR cherry-pick will not do.
You need to backport that to 2.13, merge it, rebase, then it will take effect on the next.

Thanks.

**Comment by peterzhuamazon** [comment]
https://github.com/opensearch-project/ml-commons/pull/2663

**Comment by ylwu-amzn** [comment]
> #2663

Thanks Peter, @Zhangxunmt  , let's merge this PR first, then rebase

**Review by jngz-es — APPROVED** [review]

**Review by b4sjoo — APPROVED** [review]

**Pull request merged**